### ISSUE TITLE
question/feature: determining if image has already been scanned

### ISSUE THREAD

**Issue by samcornwell** [ckotzbauer/sbom-operator#578]
I ran into an issue where I had `sbom-operator` running in my cluster, and then we changed the repo we were using to be a fresh repo, and then it proceeded to skip every image and not populate the new repo. My original assumption is that the `sbom-operator` would check the repo to see if the the image exists, and then skip it if it does.

Upon inspecting the source code, I found out the `HasAnnotation` check, and then saw in documentation that there was an `--ignore-annotation` flag. The only problem now is that it scans every image every time, which bogs down the operator.

So that leads to my question, is there a reason not to have the ability to skip the image if it's found in the repository? Shouldn't the sha265 hash guarantee that the image is an exact match in the cluster and the repository? If so, it seems the only way there would be a problem is if the `sbom.json` was manually manipulated and then committed, which seems like a case that shouldn't be considered. If I'm right about that, I was wondering if it makes sense to have the default behavior be the way I suggested, or perhaps a flag to enable it?

Or I could be wrong about the hash thing, I don't know.

Thanks

edit: IDK what the equivalent logic for the other targets would be, I'm just familiar with using the `git` target

**Comment by samcornwell** [comment]
Update: ok maybe I don't understand the logic. After running for a while with `--ignore-annotations`, it seems to have populated my repo, and now it's showing `Skip image` for everything

**Comment by ckotzbauer** [comment]
Hi @samcornwell,
to populate a new repo you have to run the operator with `--ignore-annotation` until all your sboms are created again, then you can remove the flag again. The purpose of the flag is just to force a one-time re-scan of all images. When they are scanned (and the operator did not restart) they are skipped again, when the image did not change, it uses an in-memory cache with the sha256's.

Of course it would be possible to instead always check the repo and check if the expected file is already there, but this would be much slower, this is the reason why I implemented it this way.

Does this solve your question?

**Comment by samcornwell** [comment]
Thank you. That's exactly what I tried doing, and it seemed like it was working. It seemed to scan everything. Then later I looked at my logs and saw that the most recent scans are skipping again, even though I still haven't removed the `--ignore-annotations` flag yet. I expected that it would just scan over and over until I removed the flag, because that's what it appeared to be doing at first. Is there some other check that can cause skipping?
